bgpd/bgp_vty.c

```diff
@@ -9363,6 +9364,116 @@ DEFPY(bgp_imexport_vrf, bgp_imexport_vrf_cmd,
 	return CMD_SUCCESS;
 }
 
+DEFPY(bgp_redistribute_vrf, bgp_redistribute_vrf_cmd,
+      "[no] redistribute vrf VIEWVRFNAME$import_name",
+      NO_STR
+      "Redistribute routes from another VRF\n"
+      "VRF to import from\n"
+      "The name of the VRF\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	struct listnode *node, *nnode;
+	struct listnode *exnode, *nexnode;
+	;
+	struct bgp *vrf_bgp;
+	int32_t ret = 0;
+	as_t as = bgp->as;
+	bool remove = false;
+	int32_t idx = 0;
+	char *vname;
+	enum bgp_instance_type bgp_type = BGP_INSTANCE_TYPE_VRF;
+	afi_t afi;
+
+	if (import_name == NULL) {
+		vty_out(vty, "%% Missing import name\n");
+		return CMD_WARNING;
+	}
+
+	if (strcmp(import_name, "route-map") == 0) {
+		vty_out(vty, "%% Must include route-map name\n");
+		return CMD_WARNING;
+	}
+
+	if (argv_find(argv, argc, "no", &idx))
+		remove = true;
+
+	afi = vpn_policy_getafi(vty, bgp, true);
+	if (afi == AFI_MAX)
+		return CMD_WARNING_CONFIG_FAILED;
```

Comment:
I suggest to replace these lines with something like this:

```C
afi = bgp_node_afi(vty);
safi = bgp_node_safi(vty);
if ((SAFI_UNICAST != safi) || ((AFI_IP != afi) && (AFI_IP6 != afi))) {
	vty_out(vty, "%% redistribute vrf is valid only for unicast ipv4|ipv6\n");
	return CMD_WARNING_CONFIG_FAILED;
}
```
	
and leave the `vpn_policy_getafi` function unchanged.

This should be equivalent to your implementation, but it doesn't change the `vpn_policy_getafi` function which is used in some other places.

Reply:
Done for this. Thanks for your comments.